lib/wit/gitrepo.py

```diff
@@ -242,14 +246,91 @@ def is_ancestor(self, ancestor, current=None):
         return proc.returncode == 0
 
     def repo_entries_from_commit(self, revision) -> List[RepoEntry]:
-        where = "{}:{}".format(revision, GitRepo.PKG_DEPENDENCY_FILE)
-        proc = self._git_command("show", where)
+        manifest_entries = self._read_manifest_from_commit(revision)
+        if len(manifest_entries) == 0:
+            manifest_entries = self._read_submodules_from_commit(revision)
+        return manifest_entries
+
+    def _read_manifest_from_commit(self, revision) -> List[RepoEntry]:
+        proc = self._git_command("show", "{}:{}".format(revision, GitRepo.PKG_DEPENDENCY_FILE))
         if proc.returncode:
-            log.debug("No dependency file found in repo [{}:{}]".format(revision,
+            log.debug("No wit dependency file found in repo [{}:{}]".format(revision,
                       self.path))
             return []
         return RepoEntries.parse(proc.stdout, Path(GitRepo.PKG_DEPENDENCY_FILE), revision)
 
+    def _read_submodules_from_commit(self, revision) -> List[RepoEntry]:
+        proc = self._git_command("show", "{}:{}".format(revision, GitRepo.SUBMODULE_FILE))
+        if proc.returncode:
+            log.debug("No .gitmodule file found in repo [{}:{}]".format(revision, self.path))
+            return []
+
+        log.debug("{}:{} does not have wit-manifest.json, "
+                  "reading dependencies from git submodules instead"
+                  .format(self.name, revision))
+
+        # Use the 'git config' parser to read the submodule contents.
+        # Output is of the form:
+        #     submodule.$NAME.path $PATH
+        #     submodule.$NAME.url  $REMOTE
+        proc = self._git_command("config", "-f-", "--get-regex", "submodule..*",
```

Comment:
```suggestion
        proc = self._git_command("config", "-f-", "--get-regex", r"submodule\..*",
```

For safety, we may want to explicitly escape the `.` to treat it as a literal `.`.

Reply:
Also, thinking about this regex a bit more, is there a way Git itself escapes `.` in a submodule's name? I know that it's definitely legal to include a `.` in a Git repo's name. If Git does escaping of `.`, then we'll need to unescape it when we build the name -> path/url mappings.

Reply:
It allows the dot in the submodule name, but doesn't escape it